src/ChargeTrapping/ChargeTrapping.jl

```diff
@@ -121,3 +121,85 @@ end
 
 
 
+
+"""
+    struct ConstantLifetimeChargeTrappingModel{T <: SSDFloat} <: AbstractChargeTrappingModel{T}
+        
+This constant-lifetime-based charge trapping model is similar to the Boggs model, which is constant-mean-free-path based.
+
+The model implemented has two sets of parameters for the sensitive (bulk) and inactive (dead layer/surface layer) volume respectively.
+The inactive layer effect will only be considered if the corresponding `virtual_drift_volume` is defined in the configuration file.
+
+## Fields
+* `τh::T`: Lifetime for holes in bulk volume (sensitive region) (default: `τh = 1ms`).
+* `τe::T`: Lifetime for electrons in bulk volume (default: `τe = 1ms`).
+* `τh_inactive::T`: Lifetime for holes in surface layer (dead layer/inactive layer) (default: `τh_inactive = 1ms`).
+* `τe_inactive::T`: Lifetime for electrons in surface layer (default: `τe_inactive = 1ms`).
+* `inactive_layer_geometry`: The geometry of the inactive layer. When this is defined, the model will consider the inactive effect.
+> Note: All τ must be much bigger than `Δt`.
+
+See also [Charge Trapping Models](@ref).
+"""
+struct ConstantLifetimeChargeTrappingModel{T <: SSDFloat} <: AbstractChargeTrappingModel{T} 
+    τh::T
+    τe::T
+    τh_inactive::T
+    τe_inactive::T
+    inactive_layer_geometry
+end
+
+function _calculate_signal( 
+        ctm::ConstantLifetimeChargeTrappingModel{T}, 
+        path::AbstractVector{CartesianPoint{T}}, 
+        pathtimestamps::AbstractVector{T}, 
+        charge::T,          
+        wpot::Interpolations.Extrapolation{T, 3}, 
+        S::CoordinateSystemType
+    )::Vector{T} where {T <: SSDFloat}
+
+    tmp_signal::Vector{T} = Vector{T}(undef, length(pathtimestamps))
+
+    running_sum::T = zero(T)
+    q::T = charge
+
+    inactive_layer_exist = ctm.inactive_layer_geometry !== missing
+    τ::T = ifelse(charge > 0, ctm.τh, ctm.τe)
+    τ_inactive::T = ifelse(charge > 0, ctm.τh_inactive, ctm.τe_inactive)
+
+    @inbounds for i in eachindex(tmp_signal)
+        Δt::T = (i > 1) ? (pathtimestamps[i] - pathtimestamps[i-1]) : zero(T)
+        Δq::T = (inactive_layer_exist && in(path[i], ctm.inactive_layer_geometry)) ? q * Δt/τ_inactive : q * Δt/τ
```

Comment:
Split this into two lines so it's easier to see what is happening:
```suggestion
        τi::T = (inactive_layer_exist && in(path[i], ctm.inactive_layer_geometry)) ? τ_inactive : τ
        Δq::T = q * Δt/τi
```